colossalai/communication/profiler.py

```diff
@@ -0,0 +1,270 @@
+import inspect
+import torch
+from torch.autograd.profiler import profile
+from torch.autograd.profiler_util import _format_time, _format_memory
+import torch.distributed as dist
+from torch.distributed import ReduceOp
+from colossalai.utils import get_current_device
+from typing import List
+
+
+def _get_code_location(depth: int):
+    ret = ""
+    length = len(inspect.stack())
+    for i in range(3, min(length, depth + 1)):
+        upper_frame = inspect.stack()[i]
+        function_name = inspect.stack()[i - 1].function
+        info = upper_frame.filename + "(" + str(upper_frame.lineno) + "): " + function_name + "\n"
+        ret += info
+
+    return ret
+
+
+def _format_bandwith(volme: float, time_us: int):
+    MB_PER_SECOND = (1000.0 / 1024.0) ** 2
+    mb_per_sec = volme / time_us * MB_PER_SECOND
+
+    if mb_per_sec >= 1024.0:
+        return '{:.3f}Gb/s'.format(mb_per_sec / 1024.0)
+    else:
+        return '{:.3f}Mb/s'.format(mb_per_sec)
+
+
+class CommEvent(object):
+
+    def __init__(self,
+                 count: int = 0,
+                 comm_vol: float = 0.,
+                 cuda_time: int = 0):
+        self.self_count = count
+        self.self_comm_vol = comm_vol
+        self.self_cuda_time = cuda_time
+
+    def add(self, rhs):
+        self.self_count += rhs.self_count
+        self.self_comm_vol += rhs.self_comm_vol
+        self.self_cuda_time += rhs.self_cuda_time
+
+
+class CommProf(object):
+
+    def __init__(self):
+        super().__init__()
+        self.total_count = 0
+        self.total_comm_vol = 0
+        self.total_cuda_time = 0
+        self.depth = 3

```

Comment:
pass it through parameter instead of hard coding.

Reply:
DONE